### PR TITLE
React Todo Application

### PR DESCRIPTION
**Pull request by GutemaG**
## React Todo Application

- Create class based and functional component in separate module
- Create two page (Todo and About page)
- Create hamburger menu
- Use React router

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  - [👥 Authors ](#-authors-)
+  - [🔭 Future Features ](#-future-features-)
+  - [🤝 Contributing ](#-contributing-)
+  - [⭐️ Show your support ](#️-show-your-support-)
+  - [🙏 Acknowledgments ](#-acknowledgments-)
+  - [📝 License ](#-license-)
+
+<!-- PROJECT DESCRIPTION -->
+
+# 📖 Math Magicians <a name="about-project"></a>
```
Comment (lojacho) [inline README.md]: Please correct the name of the app.

File: src/classBased/components/InputTodo.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    const { title } = this.state;
+    // eslint-disable-next-line react/prop-types
+    const { addTodoProps } = this.props;
+    if (title.trim()) {
+      addTodoProps(title);
+      this.setState({
+        title: '',
+      });
+    } else {
+      throw new Error('File somethign');
```
Comment (lojacho) [inline src/classBased/components/InputTodo.js]: Please correct this typo error

File: README.md (hunk 1 above)
Comment (GutemaG) [reply README.md]: Thank you I will do that!

File: src/classBased/components/TodoItem.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  // useEffect(() => (
+  //   () => {
+  //     console.log('Cleaning up...');
+  //   }
+  // ), []);
```
Comment (lojacho) [inline src/classBased/components/TodoItem.js]: Please don't leave pieces of code commented

File: src/functionBased/pages/About.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Link, Outlet } from 'react-router-dom';
+// import SinglePage from './SinglePage';
```
Comment (lojacho) [inline src/functionBased/pages/About.js]: Please don't leave code commented.

**Review by lojacho — CHANGES_REQUESTED** [review]
Hello,

You make a great job implementing this tutorial, please watch my comments in order to improve the best practices of coding in your project. 

Regards
Diego

**Review by lojacho — APPROVED** [review]
Good job implementing the suggestions, everything is ok !!! 
Time to review.

**Pull request merged**